src/datachain/lib/udf.py

```diff
@@ -387,8 +387,8 @@ def _prepare_rows(udf_inputs) -> "abc.Generator[Sequence[Any], None, None]":
                 result_objs = self.process_safe(udf_args)
                 udf_output = self._flatten_row(result_objs)
                 output = [{"sys__id": id_} | dict(zip(self.signal_names, udf_output))]
-                processed_cb.relative_update(1)
                 yield output
+                processed_cb.relative_update(1)
```

Comment:
We process before incrementing the callback, no? `process_safe` is what processes the udf.

Reply:
You're right, it is only related to generator. Updated the PR.